### ISSUE TITLE
add A/B testing capability

### ISSUE THREAD

**Issue by ghwood** [ietf-tools/wagtail_website#173]
It would be helpful to be able to conduct A/B (aka "split") testing on certain web pages.

There seem to be a few options for implementing this within Wagtail, with [Wagtail Experiments](https://github.com/torchbox/wagtail-experiments) being one.

**Comment by holloway** [comment]
Hi @ghwood what kinds of things would you want to test?

ie, with A/B testing often it's about measuring some outcome in order to evaluate success of A or B. What are we measuring?

**Comment by ghwood** [comment]
Hi @Holloway
A few initial conversion goals are being considered:
- meeting registrations
- endowment donations
- email list subscriptions
I expect these will be refined.

**Comment by holloway** [comment]
@ghwood thanks, and so you'd want to A/B test UI differences (eg button colours) and how they affect completion of registrations, etc?

**Comment by ghwood** [comment]
Yes. CTAs, copy, colors, etc.

**Comment by ali-haghnia** [comment]
@AndrewCalderSpringload 's recommendation for AB testing is `wagtail-ab-testing`. With our initial investigations, seems it can address all our needs.

CC: @holloway

**Comment by mgax** [comment]
> A few initial conversion goals are being considered:
> 
> * meeting registrations
> * endowment donations
> * email list subscriptions
>   I expect these will be refined.

Out of the box, `wagtail-ab-testing` can track _page visits within the site_ as conversion goals. This might mean, for example, that a user _visits_ the https://www.ietf.org/endowment/donate-ietf-endowment/ page, regardless of whether they actually make a donation.

More useful would be to track clicks on links and buttons within the website, which may navigate the user to another website, and for the conversion goals above, it's reasonably straightforward. @ghwood, if the list is up-to-date and complete, I can implement them.

Tracking conversion goals on a different website, e.g. meeting registrations, is another matter: it requires cooperation from the other website and seems out of scope for this project.

**Comment by ghwood** [comment]
Yes, this list is generally current for conversions that are clicking on links to another website:

1) initiate meeting registration -> click on a link in a Wagtail page to visit https://registration.ietf.org
2) make a donation - > successfully submit the form at https://www.ietf.org/endowment/donate-ietf-endowment/
3) email list subscriptions -> click a link to visit a webpage at https://www.ietf.org/mailman/listinfo/* (where the exact webpage would depend on the list. Let me know if I can provide any further detail about any of these.
